file-formats/aifp/type/zif_aff_aifp_v1.intf.abap

```diff
@@ -0,0 +1,223 @@
+      "! <p class="shorttext">Check Class</p>
+      "! Check class
+      check_class TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_class_check.
+  TYPES:
+    "! <p class="shorttext">Single Check</p>
+    "! Single check
+    BEGIN OF ty_single_check,
+      "! <p class="shorttext">Check Number</p>
+      "! Check number
+      "! $required
+      check_number      TYPE n LENGTH 3,
+      "! <p class="shorttext">Check Description</p>
+      "! Check description
+      check_description TYPE c LENGTH 120,
+      "! <p class="shorttext">Check Scenario</p>
+      "! Check scenario
+      check_scenario    TYPE ty_check_scenario,
+      "! <p class="shorttext">Field Check</p>
+      "! Field check
+      field_check       TYPE ty_field_check,
+      "! <p class="shorttext">Database Check</p>
+      "! Database check
+      database_check    TYPE ty_database_check,
+      "! <p class="shorttext">Class Check</p>
+      "! Class check
+      class_check       TYPE ty_class_check,
+      "! <p class="shorttext">Success Message</p>
+      "! Success message
+      success_message   TYPE ty_message,
+    END OF ty_single_check.
+  TYPES:
+  "! <p class="shorttext">Single Checks</p>
+  "! Single checks
+    ty_single_checks TYPE STANDARD TABLE OF ty_single_check WITH DEFAULT KEY.
+  TYPES:
+    "! <p class="shorttext">Check Details</p>
+    "! Check details
+    BEGIN OF ty_general_information,
+      "! <p class="shorttext">Namespace</p>
+      "! Namespace of the check
+      "! $required
+      namespace TYPE c LENGTH 15,
+      "! <p class="shorttext">AIF Check</p>
+      "! SAP Application Interface check
+      "! $required
+      aif_check TYPE c LENGTH 20,
+    END OF ty_general_information.
+  TYPES:
+    "! <p class="shorttext">Check</p>
+    "! Check
+    BEGIN OF ty_main,
+      "! $required
+      format_version          TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header                  TYPE zif_aff_types_v1=>ty_header_60_cloud,
+      "! <p class="shorttext">General Information</p>
+      "! General information
+      "! $required
+      general_information     TYPE ty_general_information,
+      "! <p class="shorttext">Error Message</p>
+      "! Error message
+      error_message           TYPE ty_message,
+      "! <p class="shorttext">Single Check Assignment</p>
+      "! Single check assignment
+      "! $required
+      single_check_assignment TYPE ty_single_checks,
+    END OF ty_main.
+
+  CONSTANTS:
+    BEGIN OF co_check_scenario,
+      "! <p class="shorttext">Simple Field Check</p>
+      "! Simple field check
+      simple     TYPE ty_check_scenario VALUE 'SIMPLE',
+      "! <p class="shorttext">Advanced Field Check</p>
+      "! Advanced field check
+      advanced   TYPE ty_check_scenario VALUE 'ADVANCED',
+      "! <p class="shorttext">Simple Field Check and Database Check</p>
+      "! Simple field check  and database check
```

Comment:
there are two spaces between "check" and "and"

Reply:
fixed with commit - [AIFP] React on AFF feedback